src/main/java/org/jitsi/protocol/xmpp/util/SimulcastGrouping.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.protocol.xmpp.util;
+
+import net.java.sip.communicator.impl.protocol.jabber.extensions.colibri.*;
+import net.java.sip.communicator.impl.protocol.jabber.extensions.jingle.*;
+
+import java.util.*;
+
+/**
+ * Utility class for the simulcast {@link SourcePacketExtension} grouping.
+ */
+public class SimulcastGrouping
+{
+    /**
+     * The list of FID(RTX) subgroups.
+     */
+    private final List<SourceGroup> fidGroups;
+
+    /**
+     * The main SIM group.
+     */
+    private final SourceGroup simGroup;
+
+    /**
+     * Extracts new {@link SimulcastGrouping} from given
+     * {@link MediaSourceGroupMap} for given root simulcast group.
+     *
+     * @param mediaType the type of the media of the SIM group.
+     * @param groupsMap the map of all groups of all media types.
+     * @param simGroup the root SIM group for which {@link SimulcastGrouping}
+     *        will be extracted.
+     *
+     * @return {@link SimulcastGrouping} which contains SIM and all FID
+     * subgroups.
+     */
+    public static SimulcastGrouping extractSimGrouping(
+            String                 mediaType,
```

Comment:
is `mediaType` necessary?  i'd think a non-video simulcast group would be an error.

Reply:
and we already pass in `simGroup` which has the simulcast group?  maybe i'm just missing what this class is trying to accomplish.  is it just to include all the FID groups which relate to the simulcast-grouped ssrcs?

Reply:
I will rework that, get rid of media type for simulcast in few other places as well